src/profiler.cpp

```diff
@@ -259,15 +258,24 @@ CodeCache* Profiler::findJvmLibrary(const char* lib_name) {
 CodeCache* Profiler::findLibraryByName(const char* lib_name) {
```

Comment:
So the new logic is - if you find a full path, return it. Otherwise, return a find by basename only. Right?

Why is it needed?

Reply:
Yes, that's right.

Agent depends on initialization of VM class methods. 
VM Init for OpenJ9 checks for presence of J9's shared library calling  `findJvmLibrary("libj9vm")`.
Before pr #2 it worked, because all library names were basename-d, identified only by the least segment of filesystem path. 
It stopped working because we changed to storing absolute pathnames of the libraries. That's why finding the J9 library by its "symbolic" name stopped working.

Reply:
I see. Okay